### PR TITLE
ci: improve GitHub action input options

### PR DESCRIPTION
**Pull request by ruancomelli**
This PR includes three main changes:

1) add new action inputs to allow better customization of Sourcery's behaviour
2) remove the `options:` input that could cause compatibility issues with the other inputs, since clients would be able to call the action with `check: false` and `options: --check`
3) add validation for the inputs to ensure that values make sense before passing them to Sourcery

Review with https://github.com/sourcery-ai/core/pull/1975.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,22 +1,44 @@
...
-      "Options passed to Sourcery. Run `sourcery review --help` to see available options."
+  target:
+    description: "File(s) or directory(ies) to review."
     required: false
-    default: "--check"
-  base_ref:
-    description: "Reference to compare the current branch with and run Sourcery on the diff. If empty, run Sourcery on the current branch."
+    default: "."
+  version:
+    description: "Sourcery CLI version, e.g. '0.12.10'. If empty, default to the latest version."
```
Comment (brendanator) [inline action.yml:18]: I wonder if this version number 0.12.10 can be automatically updated on each release?

File: action.yml [hunk 2]
```diff
@@ -40,8 +63,53 @@ runs:
...
       run: |
-        if [ "${{ inputs.base_ref }}" == "" ]; then
-          sourcery review ${{ inputs.options }} .
+
+        if [ ${{ inputs.check }} == "true" ]; then
+          check_arg="--check"
+        elif [ ${{ inputs.check }} == "false" ]; then
+          check_arg=""
+        else
+          echo "Invalid value for input 'check'"
```
Comment (brendanator) [inline action.yml:72]: I think these invalid values should fail the script?

**Review by brendanator — APPROVED** [review]
Looks great!

My main question is about input validation

File: action.yml (hunk 1 above)
Comment (ruancomelli) [reply action.yml:18]: @brendanator I quickly looked into it, but there doesn't seem to be an immediate way to do that (do you know any?)

For now, I added [a card to our product backlog](https://www.notion.so/sourcery/Automatically-update-suggested-Sourcery-version-in-the-GitHub-action-744df660397f427bb97a6792f562ea84) for us to keep track of this possible feature.

File: action.yml (hunk 2 above)
Comment (ruancomelli) [reply action.yml:71]: Done!

**Pull request merged**